src/NuGet.Clients/NuGet.Console/OutputConsole/ChannelOutputConsole.cs

```diff
@@ -0,0 +1,149 @@
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Text;
+using System.Threading;
+using Microsoft;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.RpcContracts.OutputChannel;
+using Microsoft.VisualStudio.Shell.ServiceBroker;
+using Microsoft.VisualStudio.Threading;
+using IAsyncServiceProvider = Microsoft.VisualStudio.Shell.IAsyncServiceProvider;
+using Task = System.Threading.Tasks.Task;
+
+namespace NuGetConsole
+{
+    internal class ChannelOutputConsole : BaseOutputConsole, IDisposable
+    {
+        private static readonly Encoding TextEncoding = Encoding.UTF8;
+
+        private readonly List<string> _deferredOutputMessages = new List<string>();
+        private readonly AsyncSemaphore _pipeLock = new AsyncSemaphore(1);
+
+        private Guid _channelGuid;
+        private readonly string _channelId;
+        private readonly string _channelOutputName;
+        private readonly JoinableTaskFactory _joinableTaskFactory;
+
+        private AsyncLazy<ServiceBrokerClient> _serviceBrokerClient;
+        private PipeWriter _channelPipeWriter;
+        private bool _disposedValue = false;
+
+        public ChannelOutputConsole(IAsyncServiceProvider asyncServiceProvider, Guid channelId, string outputName, JoinableTaskFactory joinableTaskFactory)
+        {
+            if (asyncServiceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(asyncServiceProvider));
+            }
+            _channelGuid = channelId;
+            _channelId = _channelGuid.ToString();
+            _channelOutputName = outputName ?? throw new ArgumentNullException(nameof(outputName));
+            _joinableTaskFactory = joinableTaskFactory ?? throw new ArgumentNullException(nameof(joinableTaskFactory));
+
+            _serviceBrokerClient = new AsyncLazy<ServiceBrokerClient>(async () =>
+            {
+                IBrokeredServiceContainer container = (IBrokeredServiceContainer)await asyncServiceProvider.GetServiceAsync(typeof(SVsBrokeredServiceContainer));
+                Assumes.Present(container);
+                IServiceBroker serviceBroker = container.GetFullAccessServiceBroker();
+                return new ServiceBrokerClient(serviceBroker, _joinableTaskFactory);
+            }, _joinableTaskFactory);
+        }
+
+        public override Task ActivateAsync()
+        {
+            // Nothing to do here in Visual Studio Online Environments mode. We do not create the pane
+            return Task.CompletedTask;
+        }
+
+        public override Task ClearAsync()
+        {
+            return CloseChannelAsync();
+        }
+
+        public override Task WriteAsync(string text)
+        {
+            return WriteToOutputChannelAsync(_channelId, _channelOutputName, text, CancellationToken.None);
```

Comment:
Is it null argument check for text needed?

Reply:
I'll add a check to skip writing if the text is null.